From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Kanaka Raju Nayana <kanaka.raju.nayana@intel.com>
Date: Fri, 10 Oct 2025 16:10:28 +0530
Subject: [PATCH] drm/xe: make drm_gpuvm dependency conditional on kernel
 version

Modify drm_gpuvm configuration to be conditional based on kernel version
compatibility. Make drm_gpuvm depend on BUILD_XE_DRM_GPUVM from compat
layer and change DRM_XE to conditionally select DRM_GPUVM only when the
backport is needed.

Signed-off-by: Kanaka Raju Nayana <kanaka.raju.nayana@intel.com>
---
 drivers/gpu/drm/xe/Kconfig | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/gpu/drm/xe/Kconfig b/drivers/gpu/drm/xe/Kconfig
index 0f3cef6..9c69a0e 100644
--- a/drivers/gpu/drm/xe/Kconfig
+++ b/drivers/gpu/drm/xe/Kconfig
@@ -2,6 +2,7 @@
 config DRM_GPUVM
 	tristate "Enable Xe-specific DRM GPUVM support"
 	depends on DRM
+	depends on BUILD_XE_DRM_GPUVM
 	default m
 	help
 	  GPU-VM representation providing helpers to manage a GPUs virtual
@@ -45,7 +46,7 @@ config DRM_XE
 	select DRM_TTM
 	select DRM_TTM_HELPER
 	select DRM_EXEC
-	select DRM_GPUVM
+	select DRM_GPUVM if BUILD_XE_DRM_GPUVM
 	select DRM_SCHED
 	select MMU_NOTIFIER
 	select WANT_DEV_COREDUMP
-- 
2.43.0

